packages/cli/src/cmds/compare/DiffLoader.ts

```diff
@@ -0,0 +1,125 @@
+import parseDiff from 'parse-diff';
+import { warn } from 'console';
+import assert from 'assert';
+
+import { executeCommand } from '../../lib/executeCommand';
+import { QueueObject, queue } from 'async';
+import { existsSync } from 'fs';
+
+export const MAX_DIFF_LENGTH = 1000 * 1000;
+
+export type Diff = {
+  command: string;
+  files: parseDiff.File[];
+};
+
+export class DiffLoaderQueue {
+  private diffQueue: QueueObject<string>;
+  private diffByRoots = new Map<string, Diff>();
+
+  constructor(public baseRevision: string, public headRevision: string) {
+    this.diffQueue = queue(this.executeDiffCommand.bind(this), 1);
+  }
+
+  async diff(sourcePathRoots: Set<string>): Promise<Diff | undefined> {
+    // Process only one diff at a time.
+
+    if (this.diffQueue.length()) await this.diffQueue.drain();
+
+    const roots = [...sourcePathRoots].sort().join(' ');
```

Comment:
What if a root contains spaces?

Reply:
I thought about this and thought, who is putting spaces in the root directory of their source tree?